### PR TITLE
Typography 

### PR DESCRIPTION
**Pull request by debscorreia**
This PR is related to Typography updates. It

- adds initial variables to text attributes (family, size, weight, render mode and letter-spacing)
- styles titles and paragraphs
- refactor Storybook's text styles
- refactor documentation's header 
- adds documentation for typography

Besides that, I also updated the color's documentation and fixed the toggle component box-sizing.

### CODE REVIEW

File: .storybook/scss/doc-header.scss [hunk 1]
```diff
@@ -1,37 +1,50 @@
...
-    max-width: 1000px;
+    max-width: $content-max-width;
     width: 100%;
     margin: 0 auto;
   }
 }
+
 .docs-title__text {
-  color: $mineral-80 !important;
+  color: $white !important;
```
Comment (rhannandrade) [inline .storybook/scss/doc-header.scss:25]: color should be the last declaration

File: .storybook/scss/doc-header.scss [hunk 2]
```diff
@@ -44,21 +57,24 @@ $docs-content-margin: $docs-title-height - $content-margin-top;
 #tab3:checked ~ .simple-tab__content.content3 { display: block; }
 
 .simple-tab + label {
-  padding: 4px 16px;
-  border-radius: 4px 4px 0 0;
+  display: inline-flex;
```
Comment (rhannandrade) [inline .storybook/scss/doc-header.scss:60]: review declaration order to group it better

File: .storybook/scss/doc-type.scss [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  &--freight {
+    font-family: $family-sans-serif;
+  }
+
+  &--ibm-plex-mono {
+    font-family: $family-monospace;
+  }
+
+  @each $name, $size in $font-sizes {
```
Comment (rhannandrade) [inline .storybook/scss/doc-type.scss:28]: Can you explain me what this line means?

File: labsystem/scss/components/_lab-toggle.scss [hunk 4]
```diff
@@ -33,6 +33,7 @@
     /* The circle that slides */
     &:before {
       position: absolute;
+      box-sizing: border-box;
```
Comment (rhannandrade) [inline labsystem/scss/components/_lab-toggle.scss:36]: the box-sizing declaration that is in `main.scss` does not work for this component?

File: .storybook/scss/doc-type.scss [hunk 5]
```diff
@@ -0,0 +1,39 @@
+.type {
+  display: block;
+  background-color: $mineral-10;
```
Comment (rhannandrade) [inline .storybook/scss/doc-type.scss:3]: visual declarations should be at the bottom

File: labsystem/scss/vendor/_storybook-overrides.scss [hunk 6]
```diff
@@ -1,21 +1,84 @@
...
 }
 
+h4.sbdocs-h4 {
+  margin-block-end: $spacing-04;
+  font-size: $font-size-7;
+  line-height: 1.14;
+}
+
+h5.sbdocs-h5 {
+  margin-block-start: $spacing-05 !important;
```
Comment (rhannandrade) [inline labsystem/scss/vendor/_storybook-overrides.scss:50]: margin-block declaration should be at the bottom

File: labsystem/scss/vendor/_storybook-overrides.scss [hunk 7]
```diff
@@ -1,21 +1,84 @@
...
+  font-weight: $weight-semibold;
+}
 
-p, li {
-  color: $black-75 !important;
-  font-family: $font-secondary !important;
-  font-size: 18px !important;
-  font-weight: 500 !important;
+h6.sbdocs-h6 {
+  margin-block-end: $spacing-04;
```
Comment (rhannandrade) [inline labsystem/scss/vendor/_storybook-overrides.scss:58]: margin-block declaration should be at the bottom

File: labsystem/scss/vendor/_storybook-overrides.scss [hunk 8]
```diff
@@ -1,21 +1,84 @@
...
+  margin-block-start: $spacing-06 !important;
+  margin-block-end: $spacing-05;
+  
+  strong {
+    font-weight: $title-strong-weight;
+  }
+}
+
+h1.sbdocs-h1 {
+  margin-block-start: $spacing-08 !important;
```
Comment (rhannandrade) [inline labsystem/scss/vendor/_storybook-overrides.scss:28]: margin-block declaration should be at the bottom

File: .storybook/scss/doc-header.scss (hunk 1 above)
Comment (debscorreia) [reply .storybook/scss/doc-header.scss]: done!

File: .storybook/scss/doc-header.scss (hunk 2 above)
Comment (debscorreia) [reply .storybook/scss/doc-header.scss:61]: done

File: .storybook/scss/doc-type.scss (hunk 5 above)
Comment (debscorreia) [reply .storybook/scss/doc-type.scss]: done

File: .storybook/scss/doc-type.scss (hunk 3 above)
Comment (debscorreia) [reply .storybook/scss/doc-type.scss:28]: Yes, sure. Let me explain in parts. `$font-sizes` is a SCSS map declared on the file `_typography.scss`. 

```
$font-sizes: (
  size-1: $font-size-1,
  size-2: $font-size-2,
  size-3: $font-size-3,
  size-4: $font-size-4,
  size-5: $font-size-5,
  size-6: $font-size-6,
  size-7: $font-size-7,
  size-8: $font-size-8,
  size-9: $font-size-9,
  size-10: $font-size-10,
  size-11: $font-size-11,
  size-12: $font-size-12,
);
```

Each line is a pair. I called them a pair of $name and $size. For example: on the first line of the map `size-1: $font-size-1,` is the pair and `$name` will be `size-1` and `$size` will be `$font-size-1`.

This piece of scss goes through each pair of `$name` and `$size` on the map `$font-sizes` and creates a css class. The class is called `&--#{$name}` and the content `font-size: $size`.

It allows us to create 12 classes, one for each font-size, with much less code.

image

File: labsystem/scss/components/_lab-toggle.scss (hunk 4 above)
Comment (debscorreia) [reply labsystem/scss/components/_lab-toggle.scss:36]: No :/ It broke again, so I added to the toggle to make sure it works.

File: labsystem/scss/vendor/_storybook-overrides.scss (hunk 8 above)
Comment (debscorreia) [reply labsystem/scss/vendor/_storybook-overrides.scss:28]: Hey, I reordered to match the declaration order we defined on the css guidelines, but I couldn't find the margin-block declaration order. I added on the same place where margin should be, but on the cases on this file is on top. Used this [link](https://gist.github.com/awkale/ad46e2ade70e833fa178) as a reference

**Pull request merged**